python/pyspark/ml/tests.py

```diff
@@ -586,10 +589,13 @@ def test_fit_maximize_metric(self):
         tvsModel = tvs.fit(dataset)
         bestModel = tvsModel.bestModel
         bestModelMetric = evaluator.evaluate(bestModel.transform(dataset))
+        validationMetrics = tvsModel.validationMetrics
 
         self.assertEqual(0.0, bestModel.getOrDefault('inducedError'),
                          "Best model should have zero induced error")
         self.assertEqual(1.0, bestModelMetric, "Best model has R-squared of 1")
+        self.assertEqual(len(grid), len(validationMetrics),
```

Comment:
Let's make this test line up with the minimize version by checking that the best model metric is the max of validation metrics.